pkg/apis/scheduling/types.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduling
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient=true
+// +nonNamespaced=true
+
+// PriorityClass defines the mapping from a priority class name to the priority
+// integer value. The value can be any valid integer.
+type PriorityClass struct {
```

Comment:
Sorry that I'm behind on reviews... trying pretty frantically to catch up.  

Why exactly do we create our own api group, and won't this need to be referenced by the main pod objects?  B/c we don't typically cross groups, they are self contained.